### PR TITLE
Master optimize setup models rco

### PR DESCRIPTION
**Pull request by rco-odoo**
Optimize `registry.setup_models()` such that some fields can be reused between registries. This happens when a model has the same base classes as another model.

This reduces the registry setup time.
On the runbot, this reduces the total execution time by ~20%.

### CODE REVIEW

**Comment by rco-odoo** [comment]
Rewrite commit history before merging.

**Comment by rco-odoo** [comment]
@KangOl @antonylesuisse can you review this? Thanks!

File: openerp/modules/registry.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ class Registry(Mapping):
 
     """
 
+    # shared cache for model classes, indexed by their base classes
+    model_cache = LRU(8192)
```
Comment (KangOl) [inline openerp/modules/registry.py]: Model not shared across registries?

**Comment by KangOl** [comment]
:ok:

**Pull request merged**